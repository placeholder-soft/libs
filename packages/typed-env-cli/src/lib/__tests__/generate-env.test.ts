import path from 'path';
import {
  generateEnvByTypedEnv,
  generateEnvNameByTypedEnv,
} from '../generate-env';

describe('call chaining', () => {
  const kSourceFilePath = path.resolve(
    __dirname,
    '../__fixtures__/usage-typed-env.ts'
  );
  const kTsconfigPath = path.resolve(
    __dirname,
    '../__fixtures__/tsconfig.t.json'
  );

  test('generate env name', () => {
    const envNames = generateEnvNameByTypedEnv({
      sourceFilePath: kSourceFilePath,
      options: {
        tsConfigFilePath: kTsconfigPath,
      },
    });
    expect(envNames).toMatchSnapshot();
  });

  test('generate env', () => {
    const env = generateEnvByTypedEnv({
      sourceFilePath: kSourceFilePath,
      options: {
        tsConfigFilePath: kTsconfigPath,
      },
    });
    expect(env).toMatchSnapshot();
  });
});

describe('assignment', () => {
  const kSourceFilePath = path.resolve(
    __dirname,
    '../__fixtures__/assignment-typed-env.ts'
  );
  const kTsconfigPath = path.resolve(
    __dirname,
    '../__fixtures__/tsconfig.t.json'
  );
  test('generate env name', () => {
    const envNames = generateEnvNameByTypedEnv({
      sourceFilePath: kSourceFilePath,
      options: {
        tsConfigFilePath: kTsconfigPath,
      },
    });
    expect(envNames).toMatchSnapshot();
  });

  test('generate env', () => {
    const env = generateEnvByTypedEnv({
      sourceFilePath: kSourceFilePath,
      options: {
        tsConfigFilePath: kTsconfigPath,
      },
    });
    expect(env).toMatchSnapshot();
  });
});
