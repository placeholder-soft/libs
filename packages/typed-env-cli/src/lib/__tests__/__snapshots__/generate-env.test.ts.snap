// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assignment generate env 1`] = `
Array [
  "aaa=",
  "ccc=",
  "eee=",
]
`;

exports[`assignment generate env name 1`] = `
Object {
  "aaa": "aaa",
  "ccc": "ccc",
  "eee": "eee",
}
`;

exports[`call chaining generate env 1`] = `
Array [
  "// default=['111','222','333']",
  "bbb='111'",
  "// default=['111','222','333']",
  "ccc='111'",
  "ddd=",
  "// default=['']",
  "eee=",
]
`;

exports[`call chaining generate env name 1`] = `
Object {
  "bbb": "bbb",
  "ccc": "ccc",
  "ddd": "ddd",
  "eee": "eee",
}
`;
